src/main/java/org/spongepowered/common/data/processor/data/entity/TameableDataProcessor.java

```diff
@@ -0,0 +1,163 @@
+            return Optional.<TameableData>of(new SpongeTameableData(uuidOptional.orNull()));
+        } else {
+            return Optional.absent();
+        }
+    }
+
+    @Override
+    public Optional<TameableData> fill(DataHolder dataHolder, TameableData manipulator, MergeFunction overlap) {
+        if(dataHolder instanceof EntityTameable) {
+            final TameableData merged = checkNotNull(overlap, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).orNull());
+            manipulator.set(Keys.TAMED_OWNER, merged.owner().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<TameableData> fill(final DataContainer container, final TameableData tameableData) {
+        tameableData.set(Keys.TAMED_OWNER, getData(container, Keys.TAMED_OWNER));
+        return Optional.of(tameableData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, TameableData manipulator, MergeFunction overlap) {
+        if(dataHolder instanceof EntityTameable) {
+            final EntityTameable entityTameable = (EntityTameable) dataHolder;
+            final DataTransactionBuilder builder = DataTransactionBuilder.builder();
+            final String sPrevTamer = entityTameable.getOwnerId();
+            final Optional<UUID> prevTamer = TameableDataProcessor.asUUID(sPrevTamer);
+            final TameableData newdata = checkNotNull(overlap, "MergeFunction must not be null").merge(from(dataHolder).orNull(), checkNotNull(manipulator));
+            final ImmutableSpongeOptionalValue<UUID> prevValue = new ImmutableSpongeOptionalValue<UUID>(Keys.TAMED_OWNER, prevTamer);
+            final ImmutableValue<Optional<UUID>> newValue = newdata.owner().asImmutable();
+            try {
+                builder.replace(prevValue);
+                entityTameable.setOwnerId(asString(newdata.owner().get()));
+                builder.success(newValue)
+                    .result(DataTransactionResult.Type.SUCCESS);
+                return builder.build();
+            } catch (Exception e) {
+                entityTameable.setOwnerId(sPrevTamer);
+                builder.reject(newValue)
+                    .result(DataTransactionResult.Type.ERROR);
+                return builder.build();
+            }
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public Optional<ImmutableTameableData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableTameableData immutable) {
+        //TODO: Health returns absent, investigate solutions.
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        //Fail to remove data, at this stage untameable tameables are not supported.
+        return DataTransactionBuilder.builder().result(DataTransactionResult.Type.FAILURE).build();
```

Comment:
Why is it unsupported? The whole point of being able to remove a `TameableData` is to "reset" the `Entity` to being "untamed".